scripts/patrol/patrol.py

```diff
@@ -793,64 +792,55 @@ def balance_hunting(self, possible_patrols: list):
-            needed_tags.append("fighting")
-            needed_tags.append("death")
-        elif chosen_tag == "injury":
-            needed_tags.append("injury")
-            needed_tags.append("blunt_force_injury")
-            needed_tags.append("big_bite_injury")
-            needed_tags.append("small_bite_injury")
-            needed_tags.append("minor_injury")
-            needed_tags.append("cold_injury")
-            needed_tags.append("hot_injury")
-        elif chosen_tag in ["less_prey", "prey"]:
-            if chosen_tag == "prey":
-                not_allowed_tag = "death"
-            prey_types = ["small_prey", "medium_prey", "large_prey", "huge_prey"]
-            for prey_type in prey_types:
-                if chosen_tag == "less_prey" and prey_type in ["large_prey", "huge_prey"]:
-                    continue
-                needed_tags.append(f"{prey_type}")
-                if prey_type != "small_prey":
-                    needed_tags.append(f"{prey_type}0")
-                    needed_tags.append(f"{prey_type}1")
-                    needed_tags.append(f"{prey_type}2")
-                    needed_tags.append(f"{prey_type}3")
-        elif chosen_tag == "more_prey":
-            not_allowed_tag = "death"
-            needed_tags.append("large_prey")
-            needed_tags.append("huge_prey")
-            needed_tags.append("huge_prey0")
-            needed_tags.append("huge_prey1")
-            needed_tags.append("huge_prey2")
-            needed_tags.append("huge_prey3")
-
-        # filter all possible patrol depending on the needed tags
-        # one of the mentioned tags should be in the patrol tag
+        # get first what kind of prey size which will be chosen
+        biome = game.clan.biome
+        season = game.clan.current_season
+        possible_prey_size = []
+        idx = 0
+        prey_size = ["very_small", "small", "medium", "large", "huge"]
+        for amount in PATROL_BALANCE[biome][season]:
+            possible_prey_size.extend(repeat(prey_size[idx],amount))
+            idx += 1
+        chosen_prey_size = choice(possible_prey_size)
+        print(f"chosen filter prey size: {chosen_prey_size}")
+
+        # filter all possible patrol depending on the needed prey size
         for patrol in possible_patrols:
-            for tag in needed_tags:
-                if tag in patrol.tags:
-                    # if there is a tag set, check if this tag is not in the current patrol
-                    if not_allowed_tag and not_allowed_tag not in patrol.tags:
-                        filtered_patrols.append(patrol)
-                        break
-                    # when there is no tag set, add the patrol
-                    elif not not_allowed_tag:
-                        filtered_patrols.append(patrol)
-                        break
+            for adaption, needed_weight in PATROL_WEIGHT_ADAPTION.items():
+                if patrol.weight in range(needed_weight[0],needed_weight[1]):
```

Comment:
For speed and clarity, I suggest changing this line to: 
`if needed_weight[0] <= patrol.weight < patrol.needed_weight[1]`

`range(stop, end)` is sometimes unclear in this context, since it's inclusive of `stop` and exclusive of `end`. This syntax makes it the exclusivity/inclusivity clear, and it faster.

Reply:
A way better way 😂 I was too focused to see the confusing thing I made.